pkg/k8s/resource_ctors.go

```diff
@@ -91,12 +99,29 @@ type CiliumResourceParams struct {
 	CRDSyncPromise promise.Promise[synced.CRDSync] `optional:"true"`
 }
 
+type ConfigParams struct {
+	cell.In
+
+	Config      Config
+	WatchConfig *WatchConfig `optional:"true"`
```

Comment:
I would prefer avoiding `optional:"true"` and forcing to explicitly provide it. It can then be just `WatchConfig` and not `*WatchConfig`.

Also consider naming it `ServiceWatchConfig` to avoid any confusion.

Reply:
We're working pretty hard to get completely rid of `pkg/option/config.go` so that is definitely not an option.